tools/ci_build/build.py

```diff
@@ -396,12 +396,11 @@ def generate_build_tree(cmake_path, source_dir, build_dir, cuda_home, cudnn_home
     build_number = os.getenv('Build_BuildNumber')
     source_version = os.getenv('Build_SourceVersion')
     if build_number and source_version:
-        build_matches = re.match(r"^(\d\d)(\d\d)(\d\d)-(\d\d)(\d\d)\.(\d)\.(\S+)$", build_number)
+        build_matches = re.match(r"^(\d\d)(\d\d)(\d\d)(\d\d)\.(\d)$", build_number)
```

Comment:
Is the last component always a single digit?

Reply:
You should glob in case the number of runs is > 9. And `re.fullmatch` matches the entire thing if you don't want to use ^$